.circleci/config.yml

```diff
@@ -159,35 +160,97 @@ jobs:
+          # We need to enable experimental features in order to use docker buildx
+          command: |
+              echo 'export DOCKER_TAG_MAIN=$(sh contrib/semver/docker.sh main)' >> $BASH_ENV
+              echo 'export DOCKER_TAG_MAJOR=$(sh contrib/semver/docker.sh major)' >> $BASH_ENV
+              echo 'export DOCKER_TAG_MAJORMINOR=$(sh contrib/semver/docker.sh majorminor)' >> $BASH_ENV
+              echo 'export DOCKER_TAG_FULL=$(sh contrib/semver/docker.sh full)' >> $BASH_ENV
+              echo 'export CINAME=$(sh contrib/semver/name.sh)' >> $BASH_ENV
+              echo 'export CIVERSION=$(sh contrib/semver/version.sh --bare)' >> $BASH_ENV
+              git config --global user.email "$(git log --format='%ae' HEAD -1)";
+              git config --global user.name "$(git log --format='%an' HEAD -1)";
+              echo 'export DOCKER_BUILDKIT=1' >> $BASH_ENV
+              sudo mkdir -p /etc/docker && echo $'{\n    "experimental": true\n}' | sudo dd of=/etc/docker/daemon.json
+              sudo systemctl restart docker
+              mkdir -p ~/.docker && echo $'{\n    "experimental": "enabled"\n}' >| ~/.docker/config.json
+              sudo curl -L -o /usr/bin/buildx https://github.com/docker/buildx/releases/download/v0.2.2/buildx-v0.2.2.linux-amd64
+              sudo chmod +x /usr/bin/buildx
+
+      - run:
+          name: Build and push docker images
+          # TODO: build armv7 binaries
+          command: |
+              mkdir -p linux/amd64 && \
+                cp /tmp/upload/${CINAME}-${CIVERSION}-linux-amd64 linux/amd64/yggdrasil
+                cp /tmp/upload/${CINAME}-${CIVERSION}-yggdrasilctl-linux-amd64 linux/amd64/yggdrasilctl
+              mkdir -p linux/386 && \
+                cp /tmp/upload/${CINAME}-${CIVERSION}-linux-i386 linux/386/yggdrasil
+                cp /tmp/upload/${CINAME}-${CIVERSION}-yggdrasilctl-linux-i386 linux/386/yggdrasilctl
+              mkdir -p linux/arm/v5 && \
+                cp /tmp/upload/${CINAME}-${CIVERSION}-linux-armel linux/arm/v5/yggdrasil
+                cp /tmp/upload/${CINAME}-${CIVERSION}-yggdrasilctl-linux-armel linux/arm/v5/yggdrasilctl
+              mkdir -p linux/arm/v6 && \
+                cp /tmp/upload/${CINAME}-${CIVERSION}-linux-armhf linux/arm/v6/yggdrasil
+                cp /tmp/upload/${CINAME}-${CIVERSION}-yggdrasilctl-linux-armhf linux/arm/v6/yggdrasilctl
+              mkdir -p linux/arm/v7 && \
+                cp /tmp/upload/${CINAME}-${CIVERSION}-linux-armhf linux/arm/v7/yggdrasil
+                cp /tmp/upload/${CINAME}-${CIVERSION}-yggdrasilctl-linux-armhf linux/arm/v7/yggdrasilctl
+              mkdir -p linux/arm64/v8 && \
+                cp /tmp/upload/${CINAME}-${CIVERSION}-linux-arm64 linux/arm64/v8/yggdrasil
+                cp /tmp/upload/${CINAME}-${CIVERSION}-yggdrasilctl-linux-arm64 linux/arm64/v8/yggdrasilctl
+              docker login -u $DOCKER_USER -p $DOCKER_PASS
+              buildx build \
+                --push \
```

Comment:
What are the prerequisites needed for pushing?

Reply:
On CircleCI variables:
- set `$DOCKER_NAMESPACE` to `yggdrasilnetwork/yggdrasil-go`
- set `$DOCKER_USER` to your (or a dummy) docker hub account user
- set `DOCKER_PASS` to the password of the account above